.github/workflows/build-and-test-arm.yml

```diff
@@ -18,7 +18,7 @@ permissions:
 
 # Do not cancel this workflow on main. See https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16616
 concurrency:
-  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  group: ${{ github.workflow }}-${{ github.sha || github.run_id }}
```

Comment:
`github.sha` is always present, so we don't need `github.run_id` right?

Reply:
sounds good, removed.